bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -663,9 +665,12 @@ void BCStateTran::saveReservedPage(uint32_t reservedPageId, uint32_t copyLength,
 
 // TODO(TK) check if this function can have its own transaction(bftimpl)
 void BCStateTran::zeroReservedPage(uint32_t reservedPageId) {
-  LOG_DEBUG(logger_, reservedPageId);
+  if (psd_->getIsFetchingState()) {
```

Comment:
`if (isCollectingState_) {`
is much faster.
I am going to make some changes (will upload PR) to make isCollectingState_ safer, although the idea of polling on the state is a very bad idea...and can always lead to wrong results.
===Same for line 648===